### PR TITLE
add GoPackage to NodeData

### PR DESCRIPTION
**Pull request by DanG100**
Add GoPackage field to path NodeData so that downstream generators can use the package name.

### CODE REVIEW

File: ypathgen/pathgen.go [hunk 1]
```diff
@@ -485,6 +486,8 @@ type NodeData struct {
 	YANGTypeName string
 	// YANGPath is the schema path of the YANG node.
 	YANGPath string
+	// GoPackage is the Go package name of a schema node.
```
Comment (wenovus) [inline ypathgen/pathgen.go:489]: `NodeData` represents both GoStruct and PathStruct information. So maybe something more specific:

"GoPathPackageName is the Go package name containing the generated PathStruct for the schema node."

File: ypathgen/pathgen.go [hunk 2]
```diff
@@ -616,9 +619,11 @@ func mustTemplate(name, src string) *template.Template {
 // schema field names of that directory entry (i.e. the same keys as the
 // "Fields" map of the Directory entry). Since ygen provides a *MappedType for
 // every leaf node only, leafTypeMap's value is nil for non-leaf nodes.
+// packageName, splitByModule, and trimOCPackage are used to generate the name of
+// Go package for each output node.
```
Comment (wenovus) [inline ypathgen/pathgen.go:623]: are used to determine the generated Go package name for the generated PathStructs.

**Review by wenovus — APPROVED** [review]

File: ypathgen/pathgen.go (hunk 1 above)
Comment (DanG100) [reply ypathgen/pathgen.go]: done

File: ypathgen/pathgen.go (hunk 2 above)
Comment (DanG100) [reply ypathgen/pathgen.go]: done

**Pull request merged**